test/object-store/results.cpp

```diff
@@ -2071,6 +2081,11 @@ TEST_CASE("notifications: results") {
                 REQUIRE_INDICES(collection_change_set_table_value.modifications, 1);
                 REQUIRE_INDICES(collection_change_set_table_value.modifications_new, 1);
 
```

Comment:
this part should not work, our intention is that modifications of objects within the result should not fire notifications at all when you provide an empty `keypatharray`